core/payment-driver/base/src/db/models.rs

```diff
@@ -28,6 +28,24 @@ pub const PAYMENT_STATUS_NOT_ENOUGH_FUNDS: i32 = 3;
 pub const PAYMENT_STATUS_NOT_ENOUGH_GAS: i32 = 4;
 pub const PAYMENT_STATUS_FAILED: i32 = 5;
 
+pub const TRANSFER_TX: i32 = 1;
+const FAUCET_TX: i32 = 0;
+
+#[derive(Clone, Copy)]
+pub enum TxType {
+    Transfer,
+    Faucet,
+}
+
+impl Into<i32> for TxType {
+    fn into(self) -> i32 {
+        match &self {
+            TxType::Transfer => TRANSFER_TX,
+            TxType::Faucet => FAUCET_TX,
+        }
+    }
+}
```

Comment:
The rust way to do this is:
```rust
pub enum TxType {
    Faucet = 0,
    Transfer = 1,
}
```

Then enum can be casted using `as i32`.

Reply:
Btw, I'm not sure about the `Faucet` variant. It's very specific to the erc-20 driver. Maybe it could be renamed to `Other` or `Internal`?

Reply:
yay for the rusty way, thanks!

the faucet variant is indeed very driver specific.. but could apply to zksync as well once they get smart contracts
i propose to make a separate task to investigate "driver specific data in a shared database schema"